src/Nethermind/Nethermind.Facade/Filters/LogFinder.cs

```diff
@@ -46,6 +50,64 @@ public class LogFinder : ILogFinder
             _logger = logManager?.GetClassLogger<LogFinder>() ?? throw new ArgumentNullException(nameof(logManager));
             _maxBlockDepth = maxBlockDepth;
             _rpcConfigGetLogsThreads = Math.Max(1, Environment.ProcessorCount / 4);
+            _logIndexStorage = logIndexStorage ?? new LogIndexStorage();
+        }
+
+        public LogFinder(IBlockFinder? blockFinder,
+            IReceiptFinder? receiptFinder,
+            IReceiptStorage? receiptStorage,
+            IBloomStorage? bloomStorage,
+            ILogManager? logManager,
+            IReceiptsRecovery? receiptsRecovery,
+            int maxBlockDepth = 1000)
+        {
+            _blockFinder = blockFinder ?? throw new ArgumentNullException(nameof(blockFinder));
+            _receiptFinder = receiptFinder ?? throw new ArgumentNullException(nameof(receiptFinder));
+            _receiptStorage = receiptStorage ?? throw new ArgumentNullException(nameof(receiptStorage)); ;
+            _bloomStorage = bloomStorage ?? throw new ArgumentNullException(nameof(bloomStorage));
+            _receiptsRecovery = receiptsRecovery ?? throw new ArgumentNullException(nameof(receiptsRecovery));
+            _logger = logManager?.GetClassLogger<LogFinder>() ?? throw new ArgumentNullException(nameof(logManager));
+            _maxBlockDepth = maxBlockDepth;
+            _rpcConfigGetLogsThreads = Math.Max(1, Environment.ProcessorCount / 4);
+            _logIndexStorage = new LogIndexStorage();
+        }
+
+        public IEnumerable<FilterLog> FindLogsTest(LogFilter filter, CancellationToken cancellationToken = default)
```

Comment:
this is temp?

Reply:
yes, to be able to create an interface similar to the existing FindLogs for easier unit testing